source/s3_client.c

```diff
@@ -724,8 +774,11 @@ struct aws_s3_meta_request *aws_s3_client_make_meta_request(
         }
     }
 
-    struct aws_byte_cursor host_header_value;
+    if (s_update_host_header_based_on_endpoint_override(client, message_headers, options->endpoint)) {
```

Comment:
trivial: maybe rename to something `apply_endpoint_override`? We don't necessarily update host header.

Reply:
nice. i like that name better. I was struggling to come up with a good name for this